### PR TITLE
Unhardcode sound and delay in PlaceBuilding.

### PR DESCRIPTION
**Pull request by phrohdoh**
https://github.com/OpenRA/OpenRA/pull/8688/files?w=1

Also stores `order.OrderString` in a variable.

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/Player/PlaceBuilding.cs [hunk 1]
```diff
@@ -15,118 +15,136 @@
...
+		{
+			this.info = info;
+		}
 
-					if (queue == null)
-						return;
+		public void ResolveOrder(Actor self, Order order)
+		{
+			var os = order.OrderString;
+			if (os != "PlaceBuilding" ||
```
Comment (cjshmyr) [inline OpenRA.Mods.Common/Traits/Player/PlaceBuilding.cs]: If you invert the comparison use `&&` instead

Comment (Phrohdoh) [inline OpenRA.Mods.Common/Traits/Player/PlaceBuilding.cs]: Oh, right. Thanks.

**Comment by Phrohdoh** [comment]
Fixed broken inverted logical check.

**Comment by GraionDilach** [comment]
Uhm, NewOptions is not a building sound, but the New Construction Options EVA announcement. The Desc is wrong.

File: OpenRA.Mods.Common/Traits/Player/PlaceBuilding.cs [hunk 2]
```diff
@@ -15,118 +15,136 @@
...
+	public class PlaceBuildingInfo : ITraitInfo
 	{
 		[Desc("Palette to use for rendering the placement sprite.")]
 		public readonly string Palette = "terrain";
+
+		[Desc("Play the NewOptionsSound this many ticks after building placement.")]
+		public readonly int NewOptionsSoundDelay = 10;
+
+		[Desc("Sound to play after building placement.")]
+		public readonly string NewOptionsSound = "NewOptions";
```
Comment (pchote) [inline OpenRA.Mods.Common/Traits/Player/PlaceBuilding.cs]: Sound &rarr; Notification here and above.

Comment (Phrohdoh) [inline OpenRA.Mods.Common/Traits/Player/PlaceBuilding.cs]: Can do.

**Comment by Phrohdoh** [comment]
@GraionDilach I never said it was a building sound.
It is a sound to be played after the build action.

**Comment by pchote** [comment]
@GraionDilach was referring to the "Sound to play after building placement." description.  This seems to imply that it is played after every building placement, but it actually only plays if the placement changes the available production options.

**Comment by Phrohdoh** [comment]
Okay, that makes sense. I'll reword that too.

**Comment by GraionDilach** [comment]
Honestly, the current wording would imply me Building->BuildSounds. Which that definitely isn't.

**Comment by abcdefg30** [comment]
:+1: / :white_check_mark: from me after the descriptions are fixed.

**Comment by Phrohdoh** [comment]
Updated the field names and descriptions.

**Comment by GraionDilach** [comment]
:+1:

**Comment by abcdefg30** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/128f27f87edf1803baa45ae7cb7ff05ab24746dc...a0745c71dc9c1acfb7545052de74a11219ea722f)

**Pull request merged**